internal/compiler/compiler.go

```diff
@@ -180,10 +181,20 @@ func (c *Compiler) ResetCache() {
 }
 
 func (c *Compiler) getSpecialVars(t *ast.Task) (map[string]string, error) {
+	entrypoint := c.Entrypoint
+	if entrypoint == "" {
+		for _, defaultName := range taskfile.DefaultTaskfiles {
+			alt := filepathext.SmartJoin(c.Dir, defaultName)
+			if _, err := os.Stat(alt); err == nil {
+				entrypoint = alt
+				break
+			}
+		}
+	}
```

Comment:
Hey @vmaerten,

Sorry for the long delay to review.

This looks mostly good, but I'd consider extracting this block into a separate function to make things cleaner. Or perhaps, we could use the existing `Exists` function if easy enough.

Reply:
No worries!
I've extracted a function and used it in both method (exists and getSpecialVar)
Way cleaner thx!